enterprise/com/src/test/java/org/neo4j/com/storecopy/FileMoveProviderTest.java

```
@@ -0,0 +1,258 @@
+        moveActions.forEach( moveToDirectory( targetDirectory ) );
+
+        // then some files are copied over the default file system
+        File expectedNormalCopy = new File( targetDirectory, aNormalFile.getName() );
+        assertTrue( expectedNormalCopy.exists() );
+
+        // and correct files are copied over the page cache
+        File expectedPageCacheCopy = new File( targetDirectory, aPageCacheFile.getName() );
+        assertTrue( expectedPageCacheCopy.toString(), pageCache.getCachedFileSystem().fileExists( expectedPageCacheCopy ) );
+    }
+
+    private Supplier<RuntimeException> failure()
+    {
+        return () -> new RuntimeException( "Fail" );
+    }
+
+    private List<File> safeList( File dir )
+    {
+        return Arrays.asList( Optional.ofNullable( dir ).map( File::listFiles ).orElse( new File[]{} ) );
+    }
+
+    private Consumer<FileMoveAction> moveToDirectory( File fileToMove )
+    {
+        return fileMoveAction -> runnableFromThrowable( () -> fileMoveAction.move( fileToMove ) ).run();
+    }
+
+    private String readFromFile( File input )
+    {
+        try
+        {
+            BufferedReader fileReader = new BufferedReader( new FileReader( input ) );
+            StringBuilder stringBuilder = new StringBuilder();
+            char[] data = new char[32];
+            int read;
+            while ( (read = fileReader.read( data )) != -1 )
+            {
+                stringBuilder.append( data, 0, read );
+            }
+            return stringBuilder.toString();
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( e );
+        }
+    }
+
+    private File createDirectory( File file )
+    {
+        runnableFromThrowable( file::mkdirs ).run();
+        return file;
+    }
+
+    private File createFile( File file )
+    {
+        runnableFromThrowable( file::createNewFile ).run();
+        return file;
+    }
```

If you just let the tests throw IOException, you don't need to do this dance with lambdas and exceptions.